### PR TITLE
fix(vercel): rename --confirm to --yes + small changes

### PR DESCRIPTION
**Pull request by Jolg42**
General cleanup I did when working on something else

- Only cache the root node_modules
- fix concurrency for resources (Vercel & co)

### CODE REVIEW

**Review by janpio — COMMENTED** [review]
Backed by docs: https://vercel.com/docs/cli/deploy#yes

We might actually switch to explicit `vercel deploy` at the same time. Will make suggestions.

File: dataproxy/vercel-edge-functions/run.sh [hunk 1]
```diff
@@ -4,4 +4,4 @@ set -eu
 
 yarn install
 
-yarn -s vercel --token=$VERCEL_TOKEN --env VERCEL_DATA_PROXY_URL="$VERCEL_DATA_PROXY_URL" --prod --scope=$VERCEL_ORG_ID --confirm --force 1> deployment-url.txt
+yarn -s vercel --token=$VERCEL_TOKEN --env VERCEL_DATA_PROXY_URL="$VERCEL_DATA_PROXY_URL" --prod --scope=$VERCEL_ORG_ID --yes --force 1> deployment-url.txt
```
Comment (janpio) [inline dataproxy/vercel-edge-functions/run.sh]:
```suggestion
yarn -s vercel deploy --token=$VERCEL_TOKEN --env VERCEL_DATA_PROXY_URL="$VERCEL_DATA_PROXY_URL" --prod --scope=$VERCEL_ORG_ID --yes --force 1> deployment-url.txt
```

Comment (Jolg42) [reply dataproxy/vercel-edge-functions/run.sh]: It's the same, but I like when it’s explicit too

File: platforms-serverless-vercel/vercel-cli/run.sh [hunk 2]
```diff
@@ -14,17 +16,16 @@ echo "VERCEL_ORG_ID: $VERCEL_ORG_ID"
 if [ "$PRISMA_CLIENT_ENGINE_TYPE" == "binary" ]; then
   echo "Binary"
   export VERCEL_PROJECT_ID=$VERCEL_API_BINARY_PROJECT_ID
```
Comment (janpio) [inline platforms-serverless-vercel/vercel-cli/run.sh:18]: # Set `binary` as default engine type
  export PRISMA_CLIENT_ENGINE_TYPE=binary

Comment (janpio) [inline platforms-serverless-vercel/vercel-cli/run.sh:18]:
```suggestion
  export VERCEL_PROJECT_ID=$VERCEL_API_BINARY_PROJECT_ID
  # Set `binary` as default engine type
  export PRISMA_CLIENT_ENGINE_TYPE=binary
```

File: platforms-serverless-vercel/vercel-cli/run.sh [hunk 3]
```diff
@@ -14,17 +16,16 @@ echo "VERCEL_ORG_ID: $VERCEL_ORG_ID"
...
   export VERCEL_PROJECT_ID=$VERCEL_API_PROJECT_ID
-  echo "VERCEL_PROJECT_ID: $VERCEL_PROJECT_ID"
-
-  yarn -s vercel --token=$VERCEL_TOKEN --build-env PRISMA_CLIENT_ENGINE_TYPE="library" --prod --scope=$VERCEL_ORG_ID --confirm --force 1> deployment-url.txt
+  # Set `libray` as default engine type
+  export PRISMA_CLIENT_ENGINE_TYPE=library
 fi
 
+echo "VERCEL_PROJECT_ID: $VERCEL_PROJECT_ID"
+yarn -s vercel --yes --force --token=$VERCEL_TOKEN --build-env PRISMA_CLIENT_ENGINE_TYPE="$PRISMA_CLIENT_ENGINE_TYPE" --scope=$VERCEL_ORG_ID 1> deployment-url.txt
```
Comment (janpio) [inline platforms-serverless-vercel/vercel-cli/run.sh]:
```suggestion
yarn -s vercel deploy --yes --force --token=$VERCEL_TOKEN --build-env PRISMA_CLIENT_ENGINE_TYPE="$PRISMA_CLIENT_ENGINE_TYPE" --scope=$VERCEL_ORG_ID 1> deployment-url.txt
```

File: platforms-serverless-vercel/vercel-cli/run.sh [hunk 4]
```diff
@@ -14,17 +16,16 @@ echo "VERCEL_ORG_ID: $VERCEL_ORG_ID"
...
-  echo "VERCEL_PROJECT_ID: $VERCEL_PROJECT_ID"
-
-  yarn -s vercel --token=$VERCEL_TOKEN --build-env PRISMA_CLIENT_ENGINE_TYPE="binary" --prod --scope=$VERCEL_ORG_ID --confirm --force 1> deployment-url.txt
 else
  echo "Library (Default)"
   export VERCEL_PROJECT_ID=$VERCEL_API_PROJECT_ID
-  echo "VERCEL_PROJECT_ID: $VERCEL_PROJECT_ID"
-
-  yarn -s vercel --token=$VERCEL_TOKEN --build-env PRISMA_CLIENT_ENGINE_TYPE="library" --prod --scope=$VERCEL_ORG_ID --confirm --force 1> deployment-url.txt
+  # Set `libray` as default engine type
```
Comment (janpio) [inline platforms-serverless-vercel/vercel-cli/run.sh]:
```suggestion
  # Set `libray` as default engine type, no matter what might be set already (except `binary`)
```

File: platforms-serverless-vercel/vercel-node-builder/run.sh [hunk 5]
```diff
@@ -11,16 +11,16 @@ echo "VERCEL_ORG_ID: $VERCEL_ORG_ID"
...
   export VERCEL_PROJECT_ID=$VERCEL_NODE_BUILDER_PROJECT_ID
-  echo "VERCEL_PROJECT_ID: $VERCEL_PROJECT_ID"
-
-  yarn -s vercel --token=$VERCEL_TOKEN --build-env DEBUG="prisma:*" --build-env PRISMA_CLIENT_ENGINE_TYPE="library" --prod --scope=$VERCEL_ORG_ID --confirm --force 1> deployment-url.txt
+  # Set `libray` as default engine type
+  export PRISMA_CLIENT_ENGINE_TYPE=library
 fi
+
+echo "VERCEL_PROJECT_ID: $VERCEL_PROJECT_ID"
+yarn -s vercel --prod --yes --force --token=$VERCEL_TOKEN --build-env DEBUG="prisma:*" --build-env PRISMA_CLIENT_ENGINE_TYPE='$PRISMA_CLIENT_ENGINE_TYPE' --scope=$VERCEL_ORG_ID 1> deployment-url.txt
```
Comment (janpio) [inline platforms-serverless-vercel/vercel-node-builder/run.sh]:
```suggestion
yarn -s vercel deploy --prod --yes --force --token=$VERCEL_TOKEN --build-env DEBUG="prisma:*" --build-env PRISMA_CLIENT_ENGINE_TYPE='$PRISMA_CLIENT_ENGINE_TYPE' --scope=$VERCEL_ORG_ID 1> deployment-url.txt
```

File: platforms-serverless-vercel/vercel-with-nextjs/run.sh [hunk 6]
```diff
@@ -10,18 +10,16 @@ echo "VERCEL_ORG_ID: $VERCEL_ORG_ID"
...
   export VERCEL_PROJECT_ID=$VERCEL_WITH_NEXTJS_PROJECT_ID
-  echo "VERCEL_PROJECT_ID: $VERCEL_PROJECT_ID"
-
-  yarn -s vercel --token=$VERCEL_TOKEN --build-env DEBUG="prisma:*" --build-env PRISMA_CLIENT_ENGINE_TYPE="library" --prod --scope=$VERCEL_ORG_ID --force --confirm 1> deployment-url.txt
+  # Set `libray` as default engine type
+  export PRISMA_CLIENT_ENGINE_TYPE=library
 fi
 
+echo "VERCEL_PROJECT_ID: $VERCEL_PROJECT_ID"
+yarn -s vercel --prod --yes --force --token=$VERCEL_TOKEN --build-env DEBUG="prisma:*" --build-env PRISMA_CLIENT_ENGINE_TYPE='$PRISMA_CLIENT_ENGINE_TYPE' --scope=$VERCEL_ORG_ID 1> deployment-url.txt
```
Comment (janpio) [inline platforms-serverless-vercel/vercel-with-nextjs/run.sh]:
```suggestion
yarn -s vercel deploy --prod --yes --force --token=$VERCEL_TOKEN --build-env DEBUG="prisma:*" --build-env PRISMA_CLIENT_ENGINE_TYPE='$PRISMA_CLIENT_ENGINE_TYPE' --scope=$VERCEL_ORG_ID 1> deployment-url.txt
```

File: platforms-serverless-vercel/vercel-with-redwood/run.sh [hunk 7]
```diff
@@ -23,17 +23,16 @@ yarn redwood deploy vercel --no-data-migrate --no-prisma
...
   export VERCEL_PROJECT_ID=$VERCEL_WITH_REDWOOD_PROJECT_ID
-  echo "VERCEL_PROJECT_ID: $VERCEL_PROJECT_ID"
-
-  yarn -s vercel --token=$VERCEL_TOKEN --env DATABASE_URL=$DATABASE_URL --build-env DEBUG="prisma:*" --build-env PRISMA_CLIENT_ENGINE_TYPE='library' --prod --scope=$VERCEL_ORG_ID --confirm --force 1> deployment-url.txt
+  # Set `libray` as default engine type
+  export PRISMA_CLIENT_ENGINE_TYPE=library
 fi
 
+echo "VERCEL_PROJECT_ID: $VERCEL_PROJECT_ID"
+yarn -s vercel --prod --yes --force --token=$VERCEL_TOKEN --env DATABASE_URL=$DATABASE_URL --build-env DEBUG="prisma:*" --build-env PRISMA_CLIENT_ENGINE_TYPE='$PRISMA_CLIENT_ENGINE_TYPE' --scope=$VERCEL_ORG_ID 1> deployment-url.txt
```
Comment (janpio) [inline platforms-serverless-vercel/vercel-with-redwood/run.sh]:
```suggestion
yarn -s vercel deploy --prod --yes --force --token=$VERCEL_TOKEN --env DATABASE_URL=$DATABASE_URL --build-env DEBUG="prisma:*" --build-env PRISMA_CLIENT_ENGINE_TYPE='$PRISMA_CLIENT_ENGINE_TYPE' --scope=$VERCEL_ORG_ID 1> deployment-url.txt
```

File: dataproxy/vercel-edge-functions/run.sh (hunk 1 above)
Comment (janpio) [reply dataproxy/vercel-edge-functions/run.sh]: Yep - I needed to double check in the docs if `--yes` also applies to the main command (and their documentation is ignoring the fact pretty much that the main command and `deploy` are the same)

Comment (Jolg42) [reply dataproxy/vercel-edge-functions/run.sh]: Did a commit

**Review by janpio — APPROVED** [review]

**Pull request merged**